pcf/particle/aws/kms/kms_key.py

```diff
@@ -0,0 +1,296 @@
+        self._set_unique_keys()
+        self.key_name = self.desired_state_definition.get('custom_config').get('key_name')
+
+    def sync_state(self):
+        """
+        Required method: Logic that updates the current state of the particle. Used in methods in
+        the base class(es).
+
+        Works by fetching the latest key information, then setting the current_state_definition
+        equal to the result. Also updates particle state before exiting.
+        """
+        status = self._get_status()
+        if not status:
+            self.state = State.terminated
+            self.current_state_definition = {}
+            return
+        # Remove the BypassPolicyLockout key from the input. Not returned by methods, cant change
+        self.desired_state_definition.pop('BypassPolicyLockoutSafetyCheck', None)
+        if 'Tags' in self.desired_state_definition:
+            status["Tags"] = self.client.list_resource_tags(KeyId=status.get('KeyId')).get('Tags')
+        if 'Policy' in self.desired_state_definition:
+            status["Policy"] = self.client.get_key_policy(KeyId=status.get('KeyId'),
+                                                          PolicyName='default')
+        self.current_state_definition = status
+        self.state = KMSKey.state_lookup.get(status.get('KeyState'))
+
+    def is_state_definition_equivalent(self):
+        """
+        Overridden method from base class (particle.py), responsible for testing if the particle
+        has reached its desired state.
+
+        Original method ran a diff_dict on the current_state_definition and the
+        desired_state_definition (A.K.A. the resource from the particle configuration). Since the
+        return of the describe_key operation is used to set the current_state_definition, and it
+        would never contain the custom_config key, it was necessary to force a filtering before the
+        comparison operation.
+
+        Returns:
+            bool
+        """
+        # Policy, Tags, BypassSafetyLockout not returned.
+        current_filtered = pcf_util.param_filter(self.current_state_definition,
+                                                 KMSKey.start_params)
+        desired_filtered = pcf_util.param_filter(self.desired_state_definition,
+                                                 KMSKey.start_params)
+        # Filtering with allowed start params means you will need to supply them in the current
+        # config if you add new possible start params.
+        if 'Policy' in desired_filtered:
+            policy_diff = DeepDiff(json.loads(desired_filtered['Policy']),
+                                   json.loads(current_filtered['Policy']), ignore_order=True)
+            if policy_diff:
+                logger.debug("Key policy is not equivalent for {0} with diff: {1}".format(
+                    self.key_name, json.dumps(policy_diff)))
+                return False  # Assumes policy can be corrected with update method
```

Comment:
since it looks like this cannot be updated now it should return true to avoid infinite loop.

Reply:
Would it perhaps make more sense to raise an error? I agree it shouldn't be an infinite loop, but I don't want to have the user falsely assume all is well with the policy. Alternatively, we could log a warning and return true as you propose.